budcamp/Homework2/Homework2/Program.cs

```diff
@@ -0,0 +1,153 @@
+
+            Task2();
+
+            Task3();
+
+            Task4();
+
+            Task5();
+
+            Task6();
+        }
+
+        private static void Task1()
+        {
+            Marker markerGreen = new Marker();
+            markerGreen.Color = "green";
+
+            Marker markerOrange = new Marker();
+            markerOrange.Color = "orange";
+
+            markerGreen.PrintMarker();
+            markerOrange.PrintMarker();
+
+            Marker? markerBlue = Marker.InitMarker(null);
+            markerBlue ??= new Marker("default");
+            markerBlue?.PrintMarker();
+        }
+
+        private static void Task2()
+        {
+            Book bookBasic = new Book();
+            bookBasic.Title = "BASIC";
+            bookBasic.Author = "S.TROELSTRA";
+
+            Book bookC = new Book();
+            bookC.Title = "C+";
+            bookC.Author = "G.RTRTG";
+
+            bookBasic.PrintBookInfo();
+            bookC.PrintBookInfo();
+        }
+
+        private static void Task3()
+        {
+            Console.WriteLine("Input the number of elements to store in the array :");
+            int arraySize = int.Parse(Console.ReadLine());
+            int[] numbers = new int[arraySize];
+
+            for(int i = 0; i < numbers.Length; i++)
+            {
+                Console.WriteLine($"pls enter your {i + 1} number");
+                numbers[i] = int.Parse(Console.ReadLine());
+
+                Console.Clear();
+            }
+
+            Console.WriteLine("The values store into the array are :");
+
+            foreach(int value in numbers)
+            {
+                Console.Write(value + " ");
+            }
+
+            Console.WriteLine();
+
+            PrintReverseValues(numbers);
+        }
+
+        public static void PrintReverseValues(int[] inputArr)
+        {
+            Array.Reverse(inputArr);
+
+            Console.WriteLine("The values store into the array in reverse are :");
+
+            foreach (int value in inputArr)
+            {
+                Console.Write(value + " ");
+            }
+        }
+
+        private static void Task4()
+        {
+            Console.WriteLine("Input the number of elements to store in the array :");
+            int arraySize = int.Parse(Console.ReadLine());
+            int[] numbers = new int[arraySize];
+
+            for (int i = 0; i < numbers.Length; i++)
+            {
+                Console.WriteLine($"pls enter your {i + 1} number");
+                numbers[i] = int.Parse(Console.ReadLine());
+
+                Console.Clear();
+            }
+
+            FindDuplicateElements(numbers);
+        }
+
+        public static void FindDuplicateElements(int[] inputArr)
+        {
+            int count = 0;
+
+            for (int i = 0; i < inputArr.Length; i++)
+            {
+                for (int j = i + 1; j < inputArr.Length; j++)
+                {
+                    if (inputArr[i] == inputArr[j])
```

Comment:
and this algorithm is not correct;
For example:
`inputArr` = [3,4,5, 4,5,4,3];
There are 3 unique values: 3,4 and 5 and they have 4 duplicates combined.
If I understood this logic this method will return 5.

Reply:
you are right.... it was not correct, but it was ownest code)